### PR TITLE
fix(moniker): Check if moniker exists before casting

### PR DESCRIPTION
**Pull request by lwander**
https://github.com/spinnaker/orca/pull/1702

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java [hunk 1]
```diff
@@ -98,7 +105,11 @@ public TaskResult execute(Stage stage) {
 
     try {
       String applicationName = (String) stage.getContext().get("application");
-      Moniker moniker = (Moniker) stage.getContext().get("moniker");
+      Moniker moniker = null;
+      if (stage.getContext().containsKey("moniker")) {
+        moniker = objectMapper.convertValue(stage.getContext().get("moniker"), Moniker.class);
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java]: The `stage.mapTo("/moniker", Moniker.class)` is more consistent with other uses IMO ... also does not require explicit injection of an `objectMapper`.

Any reason it didn't work for you?

Comment (cfieber) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java]: `stage.mapTo("/moniker", Moniker.class)`

Comment (lwander) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java]: Oh neat, will do that

Comment (lwander) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java]: I didn't know it existed - will switch

**Comment by lwander** [comment]
@andrewbackes you probably have a better handle of all the places we try to pull moniker out of the stage context - do want to take this from here?

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java (hunk 1 above)
Comment (ajordens) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/DetermineHealthProvidersTask.java]: 👍

**Comment by andrewbackes** [comment]
Yeah

**Comment by lwander** [comment]
Cool thanks Andrew

**Pull request closed without merging**